pkg/server/server.go

```diff
@@ -354,9 +357,19 @@ func (s *ProxyServer) removeFrontendsForBackendConn(agentID string, backend Back
 			ret = append(ret, frontend)
 		}
 	}
+
+	metrics.Metrics.SetProxyConnCount(s.getCount(s.frontends))
 	return ret, nil
 }
 
+func (s *ProxyServer) getCount(frontends map[string](map[int64]*ProxyClientConnection)) int {
+	count := 0
+	for _, frontend := range frontends {
+		count = count + len(frontend)
+	}
```

Comment:
This is sort of unfortunate... I guess it's not that bad since the number of agents is relatively small, but would it be better to just increment & decrement the count? I imagine this is being done for safety, but we have a small number of methods adding & removing connections. You could even extract the frontend connections to a separate object (`FrontendConnectionManager`?) if you wanted to make it really explicit.

Reply:
Both this and https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/467 touch `frontends`, and we want to rename it. If we want to use Inc/Dec, to keep accuracy I would add handling for duplicates. Maybe we can defer a rename + hygiene improvement to immediately after?

Reply:
There is a larger refactor we may want to consider: a single storage / manager in front of both pending dials and active connections, with the same mutex. That would buy some protection (until identifier re-keying) against remaining race condition.